data/currencies.json

```diff
@@ -0,0 +1,34 @@
+{
+  "USD": "USD", "DOLLAR": "USD", "BUCK": "USD", "US DOLLAR": "USD",
```

Comment:
A quick search gave this: http://stackoverflow.com/questions/2974022
Can you explore more on this? The ideal solution would have a constant time lookup for a key, while removing the duplication of values.